### PR TITLE
Homework11: Alexey Kozhan

### PR DESCRIPTION
**Pull request by AlexKozhan**
Homework11 is done in 2 python files

### CODE REVIEW

**Comment by Stml89** [comment]
Please use different filenames, not `homework1.py`, `homework2.py` and etc

**Review by Stml89 — CHANGES_REQUESTED** [review]
See comment above

**Comment by AlexKozhan** [comment]
> Please use different filenames, not `homework1.py`, `homework2.py` etc, because those files are already in the repo and git shows the differences between previous and current homework

fixed

**Comment by AlexKozhan** [comment]
> See comment above

fixed

File: deposit_2.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        into account the monthly capitalization of interest.
+        """
+        m = money
+        n = 12
+        y = duration
+        r = annual_rate
+        a = m * (1 + r / n) ** (n * y)
+        return a
+
+    def display_deposit_info(self):
```
Comment (Stml89) [inline deposit_2.py:29]: please use `__str__` method instead

File: deposit_2.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        """
+        Calculates the total deposit amount taking
+        into account the monthly capitalization of interest.
+        """
+        m = money
+        n = 12
+        y = duration
+        r = annual_rate
+        a = m * (1 + r / n) ** (n * y)
+        return a
```
Comment (Stml89) [inline deposit_2.py:27]: You do not need to create so many variables:
```suggestion
return money * (1 + annual_rate / n) ** (n * duration)
```

File: deposit_2.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        print("This class represents a deposit.")
+
+
+class Bank:
+    """
+    Represents a bank.
+    """
+
+    def __init__(self):
+        pass
```
Comment (Stml89) [inline deposit_2.py:42]: What is the purpose of this method here if it does not do anything?

File: deposit_2.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+   метод deposit принимает аргументы N и R, и
+   возвращает сумму, которая будет на счету пользователя."""
+
+
+class Deposit:
+    """
+    Represents a deposit.
+    """
+
+    def calculate_final_money(self, money: float, duration: int,
```
Comment (Stml89) [inline deposit_2.py:16]: use staticmethod here 
https://www.programiz.com/python-programming/methods/built-in/staticmethod

File: library_1.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        book.reserve(self)
+
+    def take_book(self, book):
+        """
+        Take a book.
+        """
+        book.take(self)
+
+    def return_book(self, book):
```
Comment (Stml89) [inline library_1.py:89]: use staticmethod here
https://www.programiz.com/python-programming/methods/built-in/staticmethod

File: library_1.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                  f"reserved by user {self.current_user.name}.")
+        elif self.current_user is not None:
+            print(f"Book '{self.title}' is already "
+                  f"taken by user {self.current_user.name}.")
+        else:
+            self.current_user = user
+            self.is_reserved = False
+            print(f"Book '{self.title}' is taken by user {user.name}.")
+
+    def return_book(self):
```
Comment (Stml89) [inline library_1.py:56]:
```
user1.reserve_book(book1)
user1.take_book(book1)
user2.return_book(book1)
```
Book1 was taken by user1, but user2 can return it

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: deposit_2.py (hunk 1 above)
Comment (AlexKozhan) [reply deposit_2.py]: fixed

File: deposit_2.py (hunk 2 above)
Comment (AlexKozhan) [reply deposit_2.py]: fixed, ty

File: deposit_2.py (hunk 4 above)
Comment (AlexKozhan) [reply deposit_2.py]: fixed

File: library_1.py (hunk 5 above)
Comment (AlexKozhan) [reply library_1.py]: fixed

File: library_1.py (hunk 6 above)
Comment (AlexKozhan) [reply library_1.py]: fixed, i guess

File: deposit_2.py (hunk 3 above)
Comment (AlexKozhan) [reply deposit_2.py:39]: there were linter erros when i drafted the PR, one of them stated that there is few methods (only 1) in class Deposit so there must be  at least another, so i did

**Review by Stml89 — APPROVED** [review]

**Pull request merged**